### PR TITLE
New addon: forum-toolbar

### LINKED ISSUE #3040: code button
### Is your request related to a problem? Please describe.

I want to add a code button to the forums, so that you don't have to type [code]

### Describe the solution you'd like

No.

### Describe alternatives you've considered

No.

### Additional context

I already build some code but i'm too tired to complete it now, so if someone can make a new addon then it would be great thx

```
  var inputButtonContainer = document.createElement("li");
  inputButtonContainer.className = "markItUpButton markItUpButton17";

  var inputButton = document.createElement("a");
  inputButton.id = "codeButton";

inputButton.onclick = function () {
    document.getElementById("id_body").value += "[code=" + prompt("Language") + "]" + prompt("Code") + "[/code]";
};

  inputButton.style.backgroundImage =
    "url('https://cdn.scratch.mit.edu/scratchr2/static/__9c5e4a8de50f4ce487e67140f57c59fc__/djangobb_forum/js/markitup/sets/bbcode/images/code.png')";

  inputButtonContainer.appendChild(inputButton);

  if (toolbar) {
 document.querySelector(".markItUpButton5").insertAdjacentElement("afterend", inputButtonContainer);
  }
```

### PR DESCRIPTION
**Pull request by apple502j**
Resolves #3040

### CODE REVIEW

File: addons/forum-toolbar/addon.json [hunk 1]
```diff
@@ -0,0 +1,150 @@
+{
+  "name": "Forum toolbar buttons",
+  "description": "Adds buttons in the forum editing toolbar to insert BBCode, such as [center] or [color].",
```
Comment (mxmou) [inline addons/forum-toolbar/addon.json:3]: This sounds like there are no buttons without this addon. Maybe "More forum toolbar buttons" and "... to insert more BBCode tags ..."?

File: addons/forum-toolbar/addon.json [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  ],
+  "userstyles": [
+    {
+      "url": "userstyle.css",
+      "matches": ["editingScreens"]
+    }
+  ],
+  "settings": [
+    {
+      "name": "Wikipedia",
```
Comment (mxmou) [inline addons/forum-toolbar/addon.json:19]: Instead of using a separate setting for each of these, could the "link to ..." options be items of a menu?

**Comment by CodeGuy92** [comment]
I was actually thinking about this the other day
https://greasyfork.org/en/scripts/8932-scratch-forums-composer-swag

File: addons-l10n/en/forum-toolbar.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "forum-toolbar/link": "Link to:",
+  "forum-toolbar/wp": "Wikipedia",
+  "forum-toolbar/prompt-wp": "Ttitle on Wikipedia:",
```
Comment (GrahamSH-LLK) [inline addons-l10n/en/forum-toolbar.json:4]: typo

**Comment by GrahamSH-LLK** [comment]
Could this put the cursor in the middle of the inserted tags? This is similar to what sb menu.js does

**Comment by FunctionalMetatable** [comment]
Implement #2586 as well?

**Comment by apple502j** [comment]
This currently does not deal with existing buttons.

**Comment by WorldLanguages** [comment]
TODOs:

- [ ] React to `prompt()` returning null (user cancelled)
- [ ] Focus forum post after responding to prompts
- [ ] Settings order should be: `WIKIPEDIA, SCRATCH WIKI, GOOGLE SEARCH, DICTIONARY.COM, COLOR, CENTER, CODE`
- [ ] Buttons added to toolbar by this addon are smaller than others when dark-www is enabled
- [ ] Buttons added to toolbar by this addon are smaller than others when material-forum-editor-buttons is enabled

Last 2 are weird because image-uploader does not have those problems, despite adding its own button too.

**Comment by RedGuy12** [comment]
Pls put all links in a submenu



________________________________

**Comment by WorldLanguages** [comment]
> Pls put all links in a submenu

Already the case?

**Comment by RedGuy12** [comment]
> > Pls put all links in a submenu

> Already the case?

Oh



________________________________

**Pull request merged**